### ISSUE TITLE
Add EclipseStore as a Database option to start.vaadin.com export dialogue

### ISSUE THREAD

**Issue by mshabarov** [vaadin/platform#6772]
### Describe your motivation

Re-created in Vaadin Platform, because the original issue from @LiveNathan has been moved to a private repo.

Whenever I download a project from the Vaadin app creator (start.vaadin.com), I need to manually remove all of the default database configurations and set up EclipseStore instead. This process is repetitive and time-consuming. Given that Vaadin has partnered with EclipseStore, as highlighted in the YouTube video ["Extend your pure Java experience to the persistence layer with EclipseStore"](https://www.youtube.com/live/PwR-oxV8l4U?si=ZRF4yS2c19umkYF0), it would be beneficial to offer EclipseStore as an option directly during the project setup process on start.vaadin.com.

![image](https://github.com/user-attachments/assets/53b1ff9b-9327-45b3-be24-9556fcd94108)

### Describe the solution you'd like

I would like EclipseStore to be included as a database option in the export dialogue of the Vaadin app creator. This addition would streamline the setup process for developers who prefer to use EclipseStore, aligning the setup experience with the demonstrated Vaadin and EclipseStore integration.

### Describe alternatives you've considered

Currently, the alternative involves manually removing the default database setup and configuring EclipseStore from scratch. Another alternative is to maintain a custom template project, but this lacks the flexibility and ease of use provided by the Vaadin app creator.

### Additional context

Including EclipseStore as an option would not only save time for developers but also reinforce the existing partnership between Vaadin and EclipseStore, making it easier for developers to adopt EclipseStore seamlessly within their projects.

**Comment by Artur-** [comment]
Do you have an example of what you think the end result should look like?

**Comment by LiveNathan** [comment]
Yes! Something in between the completeness of [the bookstore demo](https://github.com/eclipse-store/bookstore-demo) and the simple implementation of the [spring boot example](https://github.com/eclipse-store/store/blob/83e1210eb024a060f80eb408f3d814710ab865bf/examples/spring-boot3-simple/src/main/java/org/microstream/spring/boot/example/simple/storage/JokesStorageImpl.java)

The book store is much more real world, but I'm using the spring boot integration.

**Comment by LiveNathan** [comment]
Of course, I'm a beginner, so someone from the EclipseStore team is a better resource to define a good starting point for new projects using the Spring Boot integration.
